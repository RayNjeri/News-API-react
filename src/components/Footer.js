import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

export default class Footer extends React.Component {
  render() {
    return (
      <div className="footer">
        <Router>
          <ul>
            <li><Link to="/About Us">About Us</Link></li>
            <li><Link to='/Terms & Conditions'>Terms & Conditions</Link></li>
            <li><Link to="/Help">Help</Link></li>
            <li><Link to="/Contact Us">Contact Us</Link></li>
          </ul>
        </Router>
      </div>
    );
  }
}

