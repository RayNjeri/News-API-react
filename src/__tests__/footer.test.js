import React from 'react';
import { mount, shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Footer from '../components/Footer';

describe('Home Component', () => {
  it('Should render a footer items', () => {
    const footer = shallow(<Footer />);
    const tree = shallowToJSON(footer);
    expect(tree).toMatchSnapshot();
  });
  
  it('Should a list of items', () => {
    const expected = ['About Us', 'Help'];
    it('matches even if received contains additional elements', () => {
      expect(['About Us', 'Help', 'Contact Us']).toEqual(expect.arrayContaining(expected));
    });
  });
});