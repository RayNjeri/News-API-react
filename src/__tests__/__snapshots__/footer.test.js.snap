// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render a footer items 1`] = `
<div
  className="footer"
>
  <BrowserRouter>
    <ul>
      <li>
        <Link
          replace={false}
          to="/About Us"
        >
          About Us
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/Terms & Conditions"
        >
          Terms & Conditions
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/Help"
        >
          Help
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/Contact Us"
        >
          Contact Us
        </Link>
      </li>
    </ul>
  </BrowserRouter>
</div>
`;
